### PR TITLE
Macos implementation

### PR DESCRIPTION
**Pull request by Knightro63**
Added the ability to use this with macos.

### CODE REVIEW

File: macos/Classes/HelloPlugin.swift [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import Cocoa
+import FlutterMacOS
+
+public class HelloPlugin: NSObject, FlutterPlugin {
```
Comment (juliansteenbakker) [inline macos/Classes/HelloPlugin.swift:4]: This file can be removed

File: README.md [hunk 2]
```diff
@@ -4,7 +4,8 @@
 [![mobile_scanner](https://github.com/juliansteenbakker/nordic_dfu/actions/workflows/flutter_format.yml/badge.svg)](https://github.com/juliansteenbakker/nordic_dfu/actions/workflows/flutter_format.yml)
 [![GitHub Sponsors](https://img.shields.io/github/sponsors/juliansteenbakker?label=sponsor%20me%20)](https://github.com/sponsors/juliansteenbakker)
 
-## 5.0.0 Breaking changes!
+## 5.3.0 Breaking changes!
+From version 5.3.0, macos implimentation.
```
Comment (juliansteenbakker) [inline README.md:8]: It's not a breaking change, so we can just mention that there is support from version 6.0.0.

File: macos/Classes/NordicDfuPlugin.h [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#import <FlutterMacOS/FlutterMacOS.h>
+
```
Comment (juliansteenbakker) [inline macos/Classes/NordicDfuPlugin.h:2]: This file is not needed for macos

File: macos/Classes/NordicDfuPlugin.m [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+#import <nordic_dfu/nordic_dfu-Swift.h>
+#else
+// Support project import fallback if the generated compatibility header
+// is not copied when this plugin is created as a library.
+// https://forums.swift.org/t/swift-static-libraries-dont-copy-generated-objective-c-header/19816
+#import "nordic_dfu-Swift.h"
+#endif
+
+@implementation NordicDfuPlugin
++ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
```
Comment (juliansteenbakker) [inline macos/Classes/NordicDfuPlugin.m:12]: This file is not needed for macos

File: pubspec.yaml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 name: nordic_dfu
 description: This library allows you to do a Device Firmware Update (DFU) of your nrf51 or nrf52 chip from Nordic Semiconductor. Fork of flutter-nordic-dfu.
-version: 5.2.1
+version: 5.3.0
```
Comment (juliansteenbakker) [inline pubspec.yaml:3]: Its best to use version 6.0.0 for this change.

**Review by juliansteenbakker — CHANGES_REQUESTED** [review]
Thanks for your work! There are a few things that can be changed.

File: macos/Classes/HelloPlugin.swift (hunk 1 above)
Comment (Knightro63) [reply macos/Classes/HelloPlugin.swift]: Removed.

File: README.md (hunk 2 above)
Comment (Knightro63) [reply README.md]: Made the requested changes to the readme file.

File: macos/Classes/NordicDfuPlugin.h (hunk 3 above)
Comment (Knightro63) [reply macos/Classes/NordicDfuPlugin.h]: Removed.

File: macos/Classes/NordicDfuPlugin.m (hunk 4 above)
Comment (Knightro63) [reply macos/Classes/NordicDfuPlugin.m]: Removed.

File: pubspec.yaml (hunk 5 above)
Comment (Knightro63) [reply pubspec.yaml]: Updated to 6.0.0

**Review by juliansteenbakker — APPROVED** [review]
Thanks for your work!

**Pull request merged**